packages/database/src/migrations/20210912223650-LESMISVitalsUseEntityAggregation-modifies-data.js

```diff
@@ -0,0 +1,232 @@
+      nostu_use_t: 0,
+    },
+  },
+};
+
+// New reports
+const ENTITY_VITALS_REPORT_CONFIG = {
+  fetch: {
+    dataElements: [
+      'SDP001',
+      'nosch_total',
+      'nostu_total',
+      ...DEVELOPMENT_PARTNERS.map(dataElement => `${dataElement}_summary`),
+    ],
+    aggregations: [
+      'MOST_RECENT',
+      {
+        type: 'SUM_PER_ORG_GROUP',
+        config: {
+          dataSourceEntityType: 'sub_district',
+          aggregationEntityType: 'requested',
+        },
+      },
+    ],
+  },
+  transform: [
+    'keyValueByDataElementName',
+    {
+      transform: 'mergeRows',
+      groupBy: 'organisationUnit',
+      using: {
+        period: 'last',
+        '*': 'single',
+      },
+    },
+    {
+      transform: 'updateColumns',
+      insert: {
+        WB: '=gt($SchDP_WB_summary, 0)',
+        WC: '=gt($SchDP_WC_summary, 0)',
+        WR: '=gt($SchDP_WR_summary, 0)',
+        WV: '=gt($SchDP_WV_summary, 0)',
+        CRS: '=gt($SchDP_CRS_summary, 0)',
+        HII: '=gt($SchDP_HII_summary, 0)',
+        RtR: '=gt($SchDP_RtR_summary, 0)',
+        WFP: '=gt($SchDP_WFP_summary, 0)',
+        AEAL: '=gt($SchDP_AEAL_summary, 0)',
+        Plan: '=gt($SchDP_Plan_summary, 0)',
+        Other: '=gt($SchCVD023_summary, 0)',
+        UNICEF: '=gt($SchDP_UNICEF, 0)',
+        NumberOfSchools: '=$nosch_total',
+        NumberOfStudents: '=$nostu_total',
+        Population: '=$SDP001',
+      },
+      exclude: '*',
+    },
+  ],
+};
+
+// Add an extra data element to school vitals report
+const updateSchoolVitalsReport = async db => {
+  const report = await findSingleRecord(db, 'report', { code: 'LESMIS_school_vitals' });
+  report.config.fetch.dataElements.push('school_complete');
+  const updateIndex = report.config.transform.findIndex(x => x.transform === 'updateColumns');
+  report.config.transform[updateIndex].insert.SchoolComplete = '=$school_complete';
+};
+
+// Remove school_complete from school vitals report
+const downdateSchoolVitalsReport = async db => {
+  const report = await findSingleRecord(db, 'report', { code: 'LESMIS_school_vitals' });
+  const elementIndexToDelete = report.config.fetch.dataElements.findIndex(
+    x => x === 'school_complete',
+  );
+  report.config.fetch.dataElements.splice(elementIndexToDelete, 1);
+  const updateIndex = report.config.transform.findIndex(x => x.transform === 'updateColumns');
+  delete report.config.transform[updateIndex].insert.SchoolComplete;
+};
+
+const insertReport = async (db, { code, reportConfig, permissionGroup }) => {
+  const reportId = generateId();
+  const permissionGroupId = (
+    await findSingleRecord(db, 'permission_group', { name: permissionGroup })
+  ).id;
+  await insertObject(db, 'report', {
+    id: reportId,
+    code,
+    config: reportConfig,
+    permission_group_id: permissionGroupId,
+  });
+};
+
+exports.up = async function (db) {
+  for (const code of OBSOLETE_REPORTS) {
+    await deleteObject(db, 'report', { code });
+  }
```

Comment:
I didn't restore these in the `down` because that would involve rewriting the full reports into this migration